common/lib/xmodule/xmodule/modulestore/tests/factories.py

```diff
@@ -22,6 +22,33 @@ class Dummy(object):
     pass
 
 
+class XModuleFactoryLock(threading.local):
```

Comment:
This is here to make it obvious if someone is calling `CourseFactory` from a context where cleanup won't happen. It could have just been global, but it seemed safer to just make it thread-local under the assumption that if we do run tests multithreaded, they'll be talking to different databases.